build_linux_nvidia.sh

```diff
@@ -0,0 +1,488 @@
+  then
+    sudo rm -r ${PPLCV_DIR}/build
+  fi
+
+  # build
+  mkdir build -p && cd build
+  cmake -DHPCC_USE_CUDA=ON -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} .. && make -j${processor_num} && sudo make install
+  sudo ldconfig
+
+  # generate prebuild and pack into .tar.gz
+  if [[ $WITH_PREBUILD -eq 1 ]]
+  then
+    sudo make DESTDIR=./prebuild install
+    tar -zcvf ${WORKING_DIR}/pplcv_${PPLCV_VER}_cuda-${ARCH}-build.tar.gz -C ./prebuild/ .
+  fi
+}
+
+mmdeploy(){
+  ## mmdeploy SDK
+  echo_green "Building and installing mmdeploysdk..."
+
+  cd ${MMDEPLOY_DIR}
+
+  MMDEPLOY_DETECT_VER=$(cat mmdeploy/version.py | grep -Eo '[0-9]\.[0-9].[0-9]+')
+
+  # reinit submodules
+  git submodule update --init --recursive
+
+  # python dependencies
+  if [[ $WITH_PYTHON -eq 1 ]]
+  then
+    source ${PYTHON_VENV_DIR}/bin/activate
+
+    ## h5py (Required by mmdeploy)
+    ## h5py not directly supported by jetson and must be built/installed manually
+    sudo apt-get install pkg-config libhdf5-10* libhdf5-dev -y
+    sudo pip3 install Cython
+    sudo env H5PY_SETUP_REQUIRES=0 pip3 install -U h5py==2.9.0
+
+    pip install -e .
+  fi
+
+  # remove all build files
+  if [[ $WITH_CLEAN -eq 1 ]]
+  then
+    sudo rm -r ${MMDEPLOY_DIR}/build
+  fi
+
+  # build
+  mkdir build -p && cd build
+  cmake .. \
+    -DMMDEPLOY_BUILD_SDK=ON \
+    -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
+    -DCMAKE_CXX_COMPILER=${GCC_COMPILER} \
+    -Dpplcv_DIR=${INSTALL_PREFIX}/lib/cmake/ppl \
+    -DMMDEPLOY_TARGET_DEVICES="cuda;cpu" \
+    -DMMDEPLOY_TARGET_BACKENDS=trt \
+    -DMMDEPLOY_CODEBASES=all \
+    -DMMDEPLOY_BUILD_SDK_PYTHON_API=ON \
+    -DTENSORRT_DIR=${TENSORRT_DIR} \
+    -DCUDNN_DIR=${CUDNN_DIR}
+  cmake --build . -- -j${PROC_NUM} && sudo cmake --install .
+  sudo ldconfig
+
+  # generate prebuild and pack into .tar.gz
+  if [[ $WITH_PREBUILD -eq 1 ]]
+  then
+    sudo make DESTDIR=./prebuild install
+    tar -zcvf ${WORKING_DIR}/mmdeploysdk_${MMDEPLOY_VER}_${ARCH}-build.tar.gz -C ./prebuild/ .
+  fi
+
+  ## build mmdeploy examples
+  cp -r ${MMDEPLOY_DIR}/demo/csrc ${MMDEPLOY_DIR}/build/example
+  cd ${MMDEPLOY_DIR}/build/example
+  rm -r build
+  mkdir build -p && cd build
+  cmake -DMMDeploy_DIR=${INSTALL_PREFIX} ..
+  make all
+}
+
+all() {
+  # build all
+  prereqs
+  py_venv
+  pplcv
+  mmdeploy
+}
+
+#####
+# supported package
+package_list=(
+  "all"
+  "prereqs"
+  "py_venv"
+  "pplcv"
+  "mmdeploy"
+)
+
+#####
+# check input argument
+if contains_element "$appargument1" "${package_list[@]}"; then
+  echo_green "Build and install '$appargument1'..."
+else
+  echo_red "Unsupported argument '$appargument1'. Use one of the following:"
+  for i in ${package_list[@]}
+  do
+    echo $i
+  done
+  exit
+fi
+
+#####
+# Unattended/auto install
+if [[ $appargument2 == "auto" ]]
+then
+  WITH_UNATTENDED=1
+fi
+
+#####
+# Install dependencies in venv?
+prompt_yesno "Install misc. dependencies in the active venv? (Default:${WITH_PYTHON})"
```

Comment:
It will give me `(Default:1)\n (y/n/q)`.

Reply:
Default option is currently only used when using unattended mode. I will add automatic selection of default option when pressing ENTER, would that work for you?

Reply:
Sounds great.